lib/preview-web/src/PreviewWeb.tsx

```diff
@@ -411,129 +426,130 @@ export class PreviewWeb<TFramework extends AnyFramework> {
-        unboundStoryFn,
-      };
-      await this.renderToDOM(renderContext, element);
+    const { id, applyLoaders, unboundStoryFn, playFunction } = story;
+    let controller = createController();
 
+    const runPhase = async (phaseName: RenderPhase, asyncFn: () => MaybePromise<void>) => {
+      this.channel.emit(Events.STORY_RENDER_PHASE_CHANGED, {
+        newPhase: phaseName,
+        storyId: id,
+      });
+      await asyncFn();
       if (controller.signal.aborted) {
-        return;
-      }
-      initialRenderPhase = 'rendered';
-
-      // NOTE: if the story is torn down during the play function, there could be negative
-      // side-effects (as the play function tries to modify something that is no longer visible).
-      // In the future we will likely pass the AbortController signal into play(), and also
-      // attempt to scope the play function by passing the element.
-      //
-      // NOTE: it is possible that args/globals have changed in between us starting to render
-      // the story and executing the play function (it is also possible that they change mid-way
-      // through executing the play function). We explicitly allow such changes to re-render the
-      // story by setting `initialRenderDone=true` immediate after `renderToDOM` completes.
-      await runPlayFunction();
-      if (controller.signal.aborted) {
-        return;
+        this.channel.emit(Events.STORY_RENDER_PHASE_CHANGED, {
+          newPhase: 'aborted',
+          storyId: id,
+        });
       }
-      initialRenderPhase = 'done';
-
-      this.channel.emit(Events.STORY_RENDERED, id);
     };
 
-    // Setup a callback to run when the story needs to be re-rendered due to args or globals changes
-    // We need to be careful for race conditions if the initial rendering of the story (which
-    // can take some time due to loaders + the play function) hasn't completed yet.
-    // Our current approach is to either stop, or rerender immediately depending on which phase
-    // the initial render is in (see comments below).
-    // An alternative approach would be to *wait* until the initial render is done, before
-    // re-rendering with the new args. This would be relatively easy if we tracked the initial
-    // render via awaiting result of the call to `initialRender`. (We would also need to track
-    // if a subsequent *re-render* is in progress, but that is less likely)
-    // See also the note about cancelling below.
-    const rerenderStory = async () => {
-      // The story has not finished rendered the first time. The loaders are still running
-      // and we will pick up the new args/globals values when renderToDOM is called.
-      if (initialRenderPhase === 'init') {
-        return;
+    let loadedContext: StoryContext<TFramework>;
+    const renderStory = async ({ initial = false, forceRemount = false } = {}) => {
+      if (forceRemount) controller.abort();
+      if (controller.signal.aborted) {
```

Comment:
How is it possible for `controller.signal.aborted` to be true if we didn't just all call `abort()` in the line above?

I think this should just be one `if`